ephios/modellogging/recorders.py

```diff
@@ -182,6 +182,34 @@ def value_statements(self):
         yield {"label": self.label, "value": self.new_value}
 
 
+class RedactedModelFieldLogRecorder(ModelFieldLogRecorder):
+    """
+    Like ModelFieldLogRecorder, but redacts the value of the field.
+    Does not support foreign keys, because how would we know what to redact?
+    """
+
+    slug = "redactedmodelfield"
+
+    def attached(self, instance):
+        old = self.field.value_from_object(instance)
+        self.old_value = "***"
+        self.old_hash = hash(old)
+
+    def record(self, action: InstanceActionType, instance):
+        if self.field.one_to_one or self.field.many_to_one:
+            raise ValueError("RedactedModelFieldLogRecorder does not support foreign keys.")
+        new = self.field.value_from_object(instance)
+        self.new_value = "***"
+        self.new_hash = hash(new)
+
+    def is_changed(self):
+        return self.old_hash != self.new_hash
```

Comment:
I'm wondering whether we lack information about the old value here. You could theoretically iterate through new values trying to find one that matches the old one. The old one is lost after the first change though, so I don't think that's a problem.